apm-agent-plugins/apm-aws-sdk/apm-aws-sdk-1-plugin/src/main/java/co/elastic/apm/agent/awssdk/v1/helper/DynamoDbHelper.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package co.elastic.apm.agent.awssdk.v1.helper;
+
+import co.elastic.apm.agent.awssdk.common.AbstractDynamoDBInstrumentationHelper;
+import co.elastic.apm.agent.impl.ElasticApmTracer;
+import co.elastic.apm.agent.impl.GlobalTracer;
+import co.elastic.apm.agent.sdk.weakconcurrent.WeakConcurrent;
+import co.elastic.apm.agent.sdk.weakconcurrent.WeakMap;
+import com.amazonaws.Request;
+import com.amazonaws.http.ExecutionContext;
+
+import javax.annotation.Nullable;
+
+public class DynamoDbHelper extends AbstractDynamoDBInstrumentationHelper<Request<?>, ExecutionContext> {
+
+    static final WeakMap<Object, String> dynamoDbRequestToTableNameMap = WeakConcurrent.buildMap();
+
+    @Nullable
+    private static DynamoDbHelper INSTANCE;
+
+    public static DynamoDbHelper getInstance() {
```

Comment:
Although unlikely, there could be concurrent calls that would create multiple instances, which could produce different threads using different req-table maps. This would quickly be resolved for subsequent requests, so it is only a matter of some odd initial reqs being handled incorrectly in the worst case. So can leave it like this, especially if the lazy initialization is deliberate. An alternative is to initialize at class initialization time

Reply:
I think having the lazy initialization is good to not allocate unnecessary memory in case certain parts of the SDK are not used at all (e.g. S3 but not DynamoDB). 
Regarding race, I agree that it's possible but very unlikely. And actually we do the same in other plugins already. I think introducing a synchronized block is not worth the benefit compared to the performance impact. So I would leave it as is.